### ISSUE TITLE
Make it harder to forget to use `--release` when packaging for production

### ISSUE THREAD

**Issue by edmorley** [Malax/libcnb.rs#359]
Currently `cargo libcnb package` defaults to dev/debug builds. To enable release builds, one has to explicitly pass `--release`.

I can see users forgetting to do so when compiling/packaging their buildpacks for production, which means larger/slower buildpack binaries.

Possible options:
1. Default to release, and instead have debug builds be opt-in. Downside is this means slower builds by default, though perhaps with libcnb-test people won't be manually packaging as much?
2. Output a warning in the log output of `cargo libcnb package` when not using `--release`.
3. Have libcnb provide some other way of determining from the built assets whether they were built in debug or release mode. For example append "(Debug)" to the end of the buildpack `name` in the packaged `buildpack.toml`, or have the SBOM include the release mode (once we support SBOM) etc.

**Comment by Malax** [comment]
FWIW: It was designed this way to mimic `cargo build` behaviour which also builds `dev` by default and `release` when passed `--release`. Consistency with the rest of the ecosystem is IMO something we should not underestimate. Granted, this is not purely building but also packaging, so slightly different.

**Comment by edmorley** [comment]
Yeah I realise it mirrors `cargo build`. I think there is more potential for gotchas in our case though, since:
- there's a chance people using libcnb.rs are slightly less familiar with Rust than users of the average Rust project
- "package" potentially suggests something a little more finalised than just "build"
- the percentage of the time that `cargo libcnb package` will be used for production vs development is probably skewed slightly more towards production, if people tend to use `libcnb-test` more to test CNBs locally (though this is to be seen)

I'm not entirely sure we want to make `--release` the default (if only, because then it won't share the cross-compiled dev mode cache warmed up by `libcnb-test` when developing locally), but I think perhaps some log messaging might be worthwhile, and eventually a section in the docs about best practices for production buildpacks/publishing buildpacks etc?
